lib/src/control_panel/display_size_picker.dart

```diff
@@ -0,0 +1,233 @@
+import 'package:flutter/material.dart';
+
+import '../extensions/string_extensions.dart';
+
+/// Allows the user to input custom screen dimensions or to select from a list
+/// of popular device presets.
+class DisplaySizePicker extends StatefulWidget {
+  /// An [EnvironmentControlPanel] widget that allows the user to change the
+  /// display dimensions of the current [Scene]. A user may select from one of
+  /// several [ScreenSizePreset]s or manually enter width and height values.
+  const DisplaySizePicker({
+    super.key,
+    required this.didChangeSize,
+  });
+
+  /// Called when the width or height values are updated, either by text input
+  /// or by the user selecting one of the [ScreenSizePreset]s.
+  final void Function(num? width, num? height) didChangeSize;
```

Comment:
Why do we need to support floating point dimensions?

Reply:
I suppose we don't. I originally implemented it this way because the numbers are eventually fed into a SizedBox, which expects `double` values for `width` and `height`, but I think it makes more sense to use ints here. Will update!

Reply:
Done.